crates/rethnet_evm/benches/state_debug.rs

```diff
@@ -0,0 +1,330 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+use revm::{
+    db::StateRef,
+    primitives::{AccountInfo, Bytecode},
+};
+
+use rethnet_eth::{Address, Bytes, U256};
+use rethnet_evm::state::AccountModifierFn;
+
+mod util;
+use util::{bench_sync_state_method, prep_no_op};
+
+fn bench_account_storage_root_account_doesnt_exist(c: &mut Criterion) {
+    bench_sync_state_method(
+        c,
+        "StateDebug::account_storage_root(), account doesn't exist",
+        prep_no_op,
+        |state, number_of_accounts| {
+            let address = Address::from_low_u64_ne(number_of_accounts + 1);
+            debug_assert!(state.basic(address).unwrap().is_none());
+            let result = state.account_storage_root(&address);
+            debug_assert!(result.is_ok());
+            debug_assert!(result.unwrap().is_none());
+        },
+    );
+}
+
+fn bench_account_storage_root_account_exists(c: &mut Criterion) {
+    bench_sync_state_method(
+        c,
+        "StateDebug::account_storage_root(), account exists",
+        prep_no_op,
+        |state, number_of_accounts| {
+            let address = Address::from_low_u64_ne(number_of_accounts);
+            debug_assert!(state.basic(address).unwrap().is_some());
+            let result = state.account_storage_root(&address);
+            debug_assert!(result.is_ok());
+            debug_assert!(result.unwrap().is_some());
+        },
+    );
+}
+
+fn bench_insert_account_already_exists(c: &mut Criterion) {
+    bench_sync_state_method(
+        c,
+        "StateDebug::insert_account(), account already exists",
+        prep_no_op,
+        |mut state, number_of_accounts| {
+            let address = Address::from_low_u64_ne(number_of_accounts);
+            debug_assert!(state.basic(address).unwrap().is_some());
+            let result = state.insert_account(address, AccountInfo::default());
+            debug_assert!(result.is_ok())
+        },
+    );
+}
+
+fn bench_insert_account_doesnt_exist(c: &mut Criterion) {
```

Comment:
It would be interesting to see `insert_account_with_code` and `insert_account_without_code` as that should have a performance impact

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/3910/commits/81cee7477012ff544a29ccfe6580c75cf5ebfa60